test/unit/test_config.py

```diff
@@ -25,6 +30,47 @@ def setUp(self):
         self.prev_dir = os.getcwd()
         os.chdir(os.path.normpath(os.path.dirname(__file__) + '/../helpers'))
 
+    def test_single_config_dependency_grapher(self):
+        """ Test dependency graph for a single directory"""
+        actual_graph = networkx.DiGraph()
+        visited = []
+        current_dir = os.path.join(os.getcwd(),'mock_graph_directory/config/account_level/regional_level_2/app2')
+        single_config_dependency_grapher(
+            current_dir,
+            actual_graph,
+            visited
+        )
+
+        expected_graph = networkx.DiGraph()
+        app3 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_2/app2')
+        app1 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_1/app1')
+        expected_graph.add_node(app3)
+        expected_graph.add_node(app1)
+        expected_graph.add_edge(app1, app3)
+
+        self.assertTrue(networkx.is_isomorphic(actual_graph, expected_graph))
+
+    def test_directory_dependency_grapher(self):
+        """test dependency graph for a recursive dependency"""
+        starting_dir = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_2')
+        actual_graph = directory_dependency_grapher(starting_dir)
+
+        expected_graph = networkx.DiGraph()
+        app1 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_1/app1')
+        app2 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_2/app2')
+        app4 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_2/app4')
+        app5 = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_2/team/app5')
+        expected_graph.add_nodes_from([app1, app2, app4, app5])
+        expected_graph.add_edge(app4, app5)
+        expected_graph.add_edge(app2, app4)
+        expected_graph.add_edge(app1, app2)
+
+        self.assertTrue(networkx.is_isomorphic(actual_graph, expected_graph))
+
+    def test_has_no_depends_on(self):
+        dir = os.path.join(os.getcwd(), 'mock_graph_directory/config/account_level/regional_level_1')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'dir'](https://app.codacy.com/manual/amplify-education/terrawrap/pullRequest?prid=4739316)